onnx/defs/nn/defs.cc

```
@@ -2487,4 +2487,206 @@ ONNX_OPERATOR_SET_SCHEMA(
         )ONNX"
         ));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+  // we need at least three inputs to have a shape for this inference.
+  if (!hasNInputShapes(ctx, 3)) {
+    return;
+  }
+
+  // validating 'input' input
+  auto input_shape = ctx.getInputType(0)->tensor_type().shape();
+  if ((input_shape.dim_size() < 2) || (input_shape.dim_size() > 3)) {
+    return; // Input tensor should have two or three dimensions.
+  }
+  size_t n_input_dims = static_cast<size_t>(input_shape.dim_size());
+  if (input_shape.dim_size() > 2) {
+    // ignore first dim (batch axis) when there are three dims.
+    --n_input_dims;
+  }
+
+  // validating 'output_shape' input
+  std::vector<int64_t> output_shape;
+  const TensorProto* temp_output_shape = ctx.getInputData(1);
+  if (temp_output_shape) {
+    // input as initializer
+    auto target_shape = ParseData<int64_t>(temp_output_shape);
+    for (auto val : target_shape) {
+      output_shape.push_back(val);
+    }
+  } else {
+    // Only 2D - 3D output tensors are supported
+    if (ctx.getInputType(1)->tensor_type().shape().dim(0).dim_value() < 2 ||
+        ctx.getInputType(1)->tensor_type().shape().dim(0).dim_value() > 3) {
+      return;
+    }
+  }
+
+  // validating 'dilation' attribute
+  std::vector<int64_t> dilations;
+  if (getRepeatedAttribute(ctx, "dilation", dilations)) {
+    if (dilations.size() != n_input_dims) {
+      return;
```

same comment as above... if this is invalid then throw an error instead of silent return

---

+1. Unclear why this is marked as resolved.